fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/common/KeyStoreSerializationKeyManager.java

```diff
@@ -0,0 +1,231 @@
+        public String getKeyStoreFileName() {
+            return keyStoreFileName;
+        }
+
+        public void setKeyStoreFileName(@Nonnull String keyStoreFileName) {
+            this.keyStoreFileName = keyStoreFileName;
+        }
+
+        @Nullable
+        public String getKeyStorePassword() {
+            return keyStorePassword;
+        }
+
+        public void setKeyStorePassword(@Nonnull String keyStorePassword) {
+            this.keyStorePassword = keyStorePassword;
+        }
+
+        @Nullable
+        public String getKeyEntryPassword() {
+            return keyEntryPassword;
+        }
+
+        public void setKeyEntryPassword(@Nonnull String keyEntryPassword) {
+            this.keyEntryPassword = keyEntryPassword;
+        }
+
+        @Nullable
+        public String getDefaultKeyEntryAlias() {
+            return defaultKeyEntryAlias;
+        }
+
+        public void setDefaultKeyEntryAlias(@Nonnull String defaultKeyEntryAlias) {
+            this.defaultKeyEntryAlias = defaultKeyEntryAlias;
+        }
+
+        @Nullable
+        public List<String> getKeyEntryAliases() {
+            return keyEntryAliases;
+        }
+
+        public void setKeyEntryAliases(@Nonnull List<String> keyEntryAliases) {
+            this.keyEntryAliases = keyEntryAliases;
+        }
+
+        @Nonnull
+        public String getCipherName() {
+            return cipherName;
+        }
+
+        public void setCipherName(@Nonnull String cipherName) {
+            this.cipherName = cipherName;
+        }
+
+        @Nullable
+        public SecureRandom getSecureRandom() {
+            return secureRandom;
+        }
+
+        public void setSecureRandom(@Nonnull SecureRandom secureRandom) {
+            this.secureRandom = secureRandom;
+        }
+
+        @Nonnull
+        public KeyStoreSerializationKeyManager build() {
+            if (keyStoreFileName == null) {
+                throw new RecordCoreArgumentException("must specify key store file name");
+            }
+            if (keyStorePassword == null) {
+                keyStorePassword = "";
+            }
+            final KeyStore keyStore;
+            try {
+                keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+                try (FileInputStream fis = new FileInputStream(keyStoreFileName)) {
+                    keyStore.load(fis, keyStorePassword.toCharArray());
+                }
+            } catch (FileNotFoundException ex) {
+                throw new RecordCoreArgumentException("Key store not found", ex);
+            } catch (GeneralSecurityException | IOException ex) {
+                throw new RecordCoreArgumentException("Key store loading failed", ex);
+            }
+            if (keyEntryPassword == null) {
+                keyEntryPassword = keyStorePassword;
+            }
+            final KeyStore.ProtectionParameter keyEntryProtection = new KeyStore.PasswordProtection(keyEntryPassword.toCharArray());
```

Comment:
Yeah, probably good to move this closer

Reply:
But then something else is too far away, right? Right now, it is constructing the arguments in the order in which they are passed to the constructor.